src/site/content/en/blog/handwriting-recognition/index.md

```diff
@@ -0,0 +1,426 @@
+points later on, it makes sense to implement the point creation logic in a separate method. In the
+following example, the `addPoint()` method calculates the elapsed time from the reference timestamp.
+The temporal information is optional, but can improve recognition quality. Then, it reads the X
+and Y coordinates from the pointer event and adds the point to the current stroke.
+
+```js
+function addPoint(event) {
+  const timeElapsed = Date.now() - activeStroke.startTime;
+  activeStroke.stroke.addPoint({
+    x: event.offsetX,
+    y: event.offsetY,
+    t: timeElapsed,
+  });
+}
+```
+
+The `pointermove` event handler is called when the pointer is moved across the screen. Those points
+need to be added to the stroke as well. The event can also be raised if the pointer is not in a
+"down" state, for example when moving the cursor across the screen without pressing the mouse
+button. The event handler from the following example checks if an active stroke exists, and adds the
+new point to it.
+
+```js
+canvas.addEventListener('pointermove', (event) => {
+  if (activeStroke) {
+    addPoint(event);
+  }
+});
+```
+
+#### Recognize text
+
+When the user lifts the pointer again, you can add the stroke to your drawing by calling its
+`addStroke()` method. The following example also resets the `activeStroke`, so the `pointermove`
+handler will not add points to the completed stroke.
+
+{% Aside %} If necessary, you can also use the drawing's `getStrokes()` method to list all strokes,
+and the `removeStroke()` method to remove a particular one from the drawing. {% endAside %}
+
+Next, it's time for recognizing the user's input by calling the `getPrediction()` method on the
+drawing. Recognition usually only takes less than a few hundred milliseconds, so you can repeatedly run predictions
+if needed. The following example runs a new prediction after each completed stroke.
+
+```js
+canvas.addEventListener('pointerup', async (event) => {
+  drawing.addStroke(activeStroke.stroke);
+  activeStroke = null;
+
+  const [mostConfidentPrediction, ...lessConfidentAlternatives] = await drawing.getPrediction();
+  if (mostConfidentPrediction) {
+    console.log(mostConfidentPrediction.text);
+  }
+});
+```
+
+This method returns a promise which resolves to an array of predictions ordered by their confidence.
+The number of elements depends on the value you passed to the `alternatives` hint. You
+could use this array to present the user with a choice of possible matches, and have them select an
+option. Alternatively, you can simply go with the most confident prediction,
+which is what the developer does in the example. On the prediction object, you will find the
+recognized text.
+
+#### Detailed insights with segmentation results
+
+> TODO: Image
+
+If supported by the target platform, the prediction object can also contain a segmentation result.
+This is an array containing all recognized user-identifiable characters (`grapheme`) along
```

Comment:
```suggestion
This is an array containing all recognized user-identifiable characters (`grapheme` objects) along
```

Reply:
Technically, the elements of this array are `HandwritingSegment`s that contain a `grapheme` property. Maybe I should be more precise here.

Reply:
Now updated.